### PR TITLE
Add route for Rub el Hizb

### PR DESCRIPTION
**Pull request by talal-najam**
### Summary
 Implement #1761. Setup basic route for Rub' el Hizb. Doesn't include sidebar navigation yet as I see comments on PR #1771 that this will be discussed further.

A couple of notes/issues that might need fixing as part of this PR or raised as separate issues:
- navigating 3 or more consecutive pages results in the same page repeating - this is the case with Juz pages as well. Still investigating, might need a new issue
- may  need to wait for #1771 Hizb route PR to be merged as I believe that includes sidebar's surah/verse tracking
- playing a surah and then once it finishes, playing another surah on the same Rub el Hizb prompts to replay "undefined". Investigating how to fix this.

Appreciate any help or feedback. Thanks & JazakAllah

### Screenshots
image
image

### CODE REVIEW

**Comment by talal-najam** [comment]
Will iron out the merge/prettier issues today.

**Comment by osamasayed** [comment]
Salam @talal-najam, #1771 is merged already.

**Comment by talal-najam** [comment]
Walaikum Assalam, thanks @osamasayed . I'll push the final few changes today.

**Comment by osamasayed** [comment]
> Walaikum Assalam, thanks @osamasayed . I'll push the final few changes today.

Sounds good. JazakAlllah Khairan, btw I am resolving merge conflict so it would be easier for you. Will push in a couple of mins inshAllah.

**Comment by talal-najam** [comment]
JazakAllah Khayran Osama, just adding the chapter mapping.

**Comment by talal-najam** [comment]
The verse tracking is working fine now. I was wondering if we should rename the route / page file for Rub `src\pages\rub\[rubId]\index.tsx` to Rub el Hizb as well? Or should we keep it as Rub for simplicity

File: src/utils/validator.ts [hunk 1]
```diff
@@ -81,6 +81,24 @@ export const isValidJuzId = (juzId: string): boolean => {
   return true;
 };
 
+/**
+ * Validate a rubId which can be in-valid in 2 cases:
+ *
+ * 1. if it's a string that is not numeric e.g. "test".
+ * 2. if it's a numeric string but lies outside the range 1->30.
```
Comment (osamasayed) [inline src/utils/validator.ts:88]: nit-picking: "1->30" should be  "1->240"

**Comment by osamasayed** [comment]
Salam @talal-najam apart from my comment above, the PR looks awesome mashAllah! Once it's solved, we are good to 🚢 ! :)

**Comment by talal-najam** [comment]
Thanks Osama once again for your review and for fixing those linting/merge issues. I've changed that function's doc to 240.

**Review by osamasayed — APPROVED** [review]

**Pull request merged**